### ISSUE TITLE
Implement Fourier Approximation for Experiment 1A

### ISSUE THREAD

**Issue by mwangangibonifaceML** [mwangangibonifaceML/basisbench#9]
#Objective
Implement Fourier representation for Experiment 1A:

Temporal function approximation BenchMark.

## tasks
- [ ] create src\basisbench\models\fourier.py
- [ ] Implement a configurable maximum frequency
- [ ] Implement forward
- [ ] Add unit tests
- [ ] Test prediction shapes
- [ ] Train initial fourier model
- [ ] Evaluate on validation data and test set

## Acceptance criteria
- [ ] Maximum frequency is configurable.
- [ ] Model accepts the correct input shapes.
- [ ] Predictions have the correct shapes
- [ ] Training reduces the training loss.
- [ ] Validation and test metrics are produced.
- [ ] Existing test suite passes

## Out of scope
- MLP
- Hyperparameter search
- persistence baseline
